src/views/preview/presentation.jsx

```diff
@@ -1,332 +1,416 @@
+            studios,
+            user,
+            onFavoriteClicked,
+            onLoveClicked,
+            onSeeInside,
+            onUpdate
+            // ...otherProps TBD
+        } = this.props;
+        const shareDate = (projectInfo.history && projectInfo.history.shared) ? projectInfo.history.shared : '';
+        return (
+            <div className="preview">
+                {projectInfo.history && shareDate === '' &&
+                    <ShareBanner>
                         <FlexRow className="preview-row">
-                            <FlexRow className="project-header">
-                                <Avatar
-                                    src={`https://cdn2.scratch.mit.edu/get_image/user/${projectInfo. author.id}_48x48.png`}
-                                />
-                                <div className="title">
-                                    {editable ?
-                                        
-                                        <InplaceInput
-                                            className="project-title"
-                                            handleUpdate={onUpdate}
-                                            name="title"
-                                            validationErrors={{
-                                                maxLength: 'Sorry title is too long'
-                                                // maxLength: props.intl.formatMessage({
-                                                //     id: 'project.titleMaxLength'
-                                                // })
-                                            }}
-                                            validations={{
-                                                // TODO: actual 100
-                                                maxLength: 40
-                                            }}
-                                            value={projectInfo.title}
-                                        /> :
-                                        <div className="project-title">{projectInfo.title}</div>
-                                    }
-                                    {`${intl.formatMessage({id: 'thumbnail.by'})} `}
+                            <span className="share-text">
+                                This project is not shared — so only you can see it. Click share to let everyone see it!
+                            </span>
+                            <Button className="button share-button">
+                                Share
+                            </Button>
+                        </FlexRow>
+                    </ShareBanner>
+                }
+                
+                { projectInfo && projectInfo.author && projectInfo.author.id && (
+                    <div className="inner">
+                        <Formsy>
+                            <FlexRow className="preview-row">
+                                <FlexRow className="project-header">
```

Comment:
This component is getting pretty large and unwieldy, with lots of pieces displaying/hiding dependent on props, and lots of nesting. I think it might be good to break some of these out into smaller components, and try to structure this so that the main component mostly contains show/hide logic based on the props, and then the sub-components just output what they're given with less logic in them.  Some of those components have nested logic, which you could manage from the main Preview component by passing the pieces in with props.  I think it would make this easier to read.

E.g., something closer to
```jsx
<div classname="preview">
    { /* Doesn't display if shareBannerProps are undefined, etc. */ }
    <ShareBanner {...shareBannerProps} />
    <ProjectMetadata
        // Displays with editable field if appropriate
        renderTitle={editable ?
            <ProjectTitle {...projectTitleProps} /> :
            <EditableProjectTitle {...projectTitleProps} />
        }
        // ...
     />
    { /* ... */ }
</div>
```